src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1473,6 +1473,20 @@ protected override void InvokeDefaultAction(string path)
             return true;
         }
 
+        private bool DetectRecursiveSymLink(DirectoryInfo path)
```

Comment:
I think `IsRecursiveSymLink` would be a better name for this method. We use `Is*` and `Has*` for methods of similar purposes in other places.